Casks/puppet-agent.rb

```diff
@@ -1,28 +1,25 @@
 cask 'puppet-agent' do
-  os_ver = MacOS.version.to_s
-  
-  if MacOS.version == :yosemite
-    version '5.5.0-1'
+  case MacOS.version
+  when '10.10'
+    version '5.5.0'
     sha256 '3f30c36e9b39763839148aaea400193c7b52d8feea2765121f6dabace658ec25'
-  elsif MacOS.version == :el_capitan
-    version '5.5.0-1'
+  when '10.11'
```

Comment:
@bmjen I forget if there was a reason we were using the names instead of version numbers? But I think there was a reason...

Reply:
I should probably try this out on the different OS versions to double check they work.

Reply:
MacOS.version behaves in interesting ways; it's default representation seems to be the version number (as a string), to get the code name as a symbol you can call `MacOS.version.to_sym`. I felt the version number, used consistently throughout the cask, was clearer.

Reply:
Definitely - I initially used it as well and recall there being a reason not to. But I don't recall the reason or if it applies here.

Reply:
Seems to work running through a few examples

### 10.10
```
$ curl http://pl-build-tools.delivery.puppetlabs.net/osx/homebrew_sierra.tar.gz | tar -x --strip 1 -C /usr/local -f -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2059k  100 2059k    0     0  8378k      0 --:--:-- --:--:-- --:--:-- 8406k
$ brew cask install puppet-agent.rb
==> Tapping caskroom/cask
Cloning into '/usr/local/Library/Taps/caskroom/homebrew-cask'...
remote: Counting objects: 4126, done.
remote: Compressing objects: 100% (4112/4112), done.
remote: Total 4126 (delta 25), reused 822 (delta 11), pack-reused 0
Receiving objects: 100% (4126/4126), 1.30 MiB | 0 bytes/s, done.
Resolving deltas: 100% (25/25), done.
Checking connectivity... done.
Tapped 0 formulae (4,134 files, 4.1M)
==> Creating Caskroom at /usr/local/Caskroom
==> Satisfying dependencies
complete
==> Downloading https://downloads.puppet.com/mac/puppet5/10.10/x86_64/puppet-agent-5.5.0-1.osx10.10.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask puppet-agent
==> Running installer for puppet-agent; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is puppet-agent
==> installer: Installing at base path /
==> installer: The install was successful.
🍺  puppet-agent was successfully installed!
```

### 10.11
```
$ curl http://pl-build-tools.delivery.puppetlabs.net/osx/homebrew_sierra.tar.gz | tar -x --strip 1 -C /usr/local -f -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2059k  100 2059k    0     0  2552k      0 --:--:-- --:--:-- --:--:-- 2552k
$ brew cask install puppet-bolt.rb
==> Tapping caskroom/cask
Cloning into '/usr/local/Library/Taps/caskroom/homebrew-cask'...
remote: Counting objects: 4126, done.
remote: Compressing objects: 100% (4112/4112), done.
remote: Total 4126 (delta 25), reused 822 (delta 11), pack-reused 0
Receiving objects: 100% (4126/4126), 1.30 MiB | 0 bytes/s, done.
Resolving deltas: 100% (25/25), done.
Checking connectivity... done.
Tapped 0 formulae (4,134 files, 4.0M)
==> Creating Caskroom at /usr/local/Caskroom
==> Satisfying dependencies
complete
==> Downloading https://downloads.puppet.com/mac/puppet5/10.11/x86_64/puppet-bolt-0.21.6-1.osx10.11.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask puppet-bolt
==> Running installer for puppet-bolt; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is puppet-bolt
==> installer: Installing at base path /
==> installer: The install was successful.
🍺  puppet-bolt was successfully installed!
```

### 10.12
```
$ curl http://pl-build-tools.delivery.puppetlabs.net/osx/homebrew_sierra.tar.gz | tar -x --strip 1 -C /usr/local -f -
$ brew cask install puppet-bolt.rb
==> Tapping caskroom/cask
Cloning into '/usr/local/Library/Taps/caskroom/homebrew-cask'...
remote: Counting objects: 4126, done.
remote: Compressing objects: 100% (4112/4112), done.
remote: Total 4126 (delta 25), reused 822 (delta 11), pack-reused 0
Receiving objects: 100% (4126/4126), 1.30 MiB | 0 bytes/s, done.
Resolving deltas: 100% (25/25), done.
Checking connectivity... done.
Tapped 0 formulae (4,134 files, 4.1M)
==> Creating Caskroom at /usr/local/Caskroom
==> Satisfying dependencies
complete
==> Downloading https://downloads.puppet.com/mac/puppet5/10.12/x86_64/puppet-bolt-0.21.6-1.osx10.12.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask puppet-bolt
==> Running installer for puppet-bolt; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is puppet-bolt
==> installer: Installing at base path /
==> installer: The install was successful.
🍺  puppet-bolt was successfully installed!
```